dashboard/lib/api/v1/school_autocomplete.rb

```diff
@@ -0,0 +1,54 @@
+require 'singleton'
```

Comment:
I'm not familiar with the `dashboard/lib/api/v1` path-convention, but I assume there's a pattern being followed in moving this class-file here?

Reply:
Yes. There are not many tests here but I was following the convention used for PD.

Reply:
The `api/v1` namespace is for libs related to the `api/v1` routes. This probably doesn't belong there.

Reply:
Never mind. I didn't realize the controller was an Api/V1 controller. This does belong here :)

However I expect this path / class name mismatch to break the autoloading. It should be `Api::V1::SchoolAutocomplete` to be found (under the [autoload root dashboard/lib](https://github.com/code-dot-org/code-dot-org/blob/1147ce6045918cb864279613c4b5e893aa812a2f/dashboard/config/application.rb#L117)) in `api/v1/school_autocomplete.rb'

As it stands I think you may have undefined class errors when calling the controller.